integrations/github/src/misc/error-handling.ts

```diff
@@ -0,0 +1,20 @@
+import * as sdk from '@botpress/sdk'
+
+/**
+ * Wraps a function with a try-catch block that catches any errors and logs them
+ * as a `sdk.RuntimeError`.
+ *
+ * @param fn - The function to wrap
+ * @param errorMessage - The error message to log
+ * @returns A function identical to the input function, but wrapped with a try-catch block
+ */
+export const wrapWithTryCatch = <T extends (...args: any[]) => Promise<any>>(fn: T, errorMessage: string): T => {
```

Comment:
**question:** why not include this with the `createActionWrapper` ?

Reply:
Some integrations might want to expose the full error message, while some might want to redact it because it contains sensitive data. I don't think there's a one-size-fits-all solution when it comes to error handling unfortunately. This could however be remediated by using an api façade and do the error redaction in there, re-throwing the redacted error, which could then safely be handled by this function by throwing a RuntimeError. This, however, would rely on us trusting that, in all integrations, errors are always being caught and re-thrown in a redacted form, which could come back to bite us in the future if we forget to do so.